tools/ossf_best_practices/passing_criteria.md

```diff
@@ -0,0 +1,397 @@
+
+**Met**
+all crypto uses openssl
+
+> All functionality in the software produced by the project that depends on cryptography MUST be implementable using FLOSS.
+
+**Met**
+
+> The security mechanisms within the software produced by the project MUST use default keylengths that at least meet the NIST minimum requirements through the year 2030 (as stated in 2012). It MUST be possible to configure the software so that smaller keylengths are completely disabled.
+
+**Met**
+
+> The default security mechanisms within the software produced by the project MUST NOT depend on broken cryptographic algorithms (e.g., MD4, MD5, single DES, RC4, Dual_EC_DRBG), or use cipher modes that are inappropriate to the context, unless they are necessary to implement an interoperable protocol (where the protocol implemented is the most recent version of that standard broadly supported by the network ecosystem, that ecosystem requires the use of such an algorithm or mode, and that ecosystem does not offer any more secure alternative). The documentation MUST describe any relevant security risks and any known mitigations if these broken algorithms or modes are necessary for an interoperable protocol.
+
+**Met**
+
+
+> The default security mechanisms within the software produced by the project SHOULD NOT depend on cryptographic algorithms or modes with known serious weaknesses (e.g., the SHA-1 cryptographic hash algorithm or the CBC mode in SSH).
+
+**Met**
+
+> The security mechanisms within the software produced by the project SHOULD implement perfect forward secrecy for key agreement protocols so a session key derived from a set of long-term keys cannot be compromised if one of the long-term keys is compromised in the future.
+
+**Met**
+
+> If the software produced by the project causes the storing of passwords for authentication of external users, the passwords MUST be stored as iterated hashes with a per-user salt by using a key stretching (iterated) algorithm (e.g., Argon2id, Bcrypt, Scrypt, or PBKDF2). See also OWASP Password Storage Cheat Sheet.
+
+**N/A**
+
+> The security mechanisms within the software produced by the project MUST generate all cryptographic keys and nonces using a cryptographically secure random number generator, and MUST NOT do so using generators that are cryptographically insecure.
+
+**Met**
+
+## Secured delivery against man-in-the-middle (MITM) attacks
+
+> The project MUST use a delivery mechanism that counters MITM attacks. Using https or ssh+scp is acceptable.
+
+**Met**
+
+> A cryptographic hash (e.g., a sha1sum) MUST NOT be retrieved over http and used without checking for a cryptographic signature.
+
+**Met**
+https://github.com/nodejs/node#verifying-binaries
+
+
+## Publicly known vulnerabilities fixed
+
+> There MUST be no unpatched vulnerabilities of medium or higher severity that have been publicly known for more than 60 days.
```

Comment:
This is also an interesting one for discussion.

Reply:
Good one! I created this issue https://github.com/nodejs/security-wg/issues/986